packages/circuit-ui/components/ColorInput/ColorInput.tsx

```diff
@@ -106,8 +105,35 @@ export const ColorInput = forwardRef<InputElement, ColorInputProps>(
 
     const hasSuffix = Boolean(suffix);
 
+    const handlePaste: ClipboardEventHandler<InputElement> = (e) => {
+      if (
+        !colorPickerRef.current ||
+        !colorInputRef.current ||
+        disabled ||
+        readOnly
+      ) {
+        return;
+      }
+
+      e.preventDefault();
+
+      const pastedText = e.clipboardData.getData('text/plain').trim();
+
+      if (!pastedText || !/^#[0-9A-F]{6}$/i.test(pastedText)) {
```

Comment:
```suggestion
      if (!pastedText || !/^#?[0-9A-F]{3,6}$/i.test(pastedText)) {
```

Reply:
I think the `#` should be optional, but then you also need to add it when setting the value of the color picker input.

Reply:
Adjusted for pasting without `#`. Allowing 3-char colors is a little bit more tricky because html color input only support 7-character strings (6 for color + `#`)[^1].

[^1]: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/color